src/orderly/run.py

```diff
@@ -56,6 +62,36 @@ def _validate_src_directory(name, root):
     return path
 
 
+def _validate_parameters(given, defaults):
+    if given is None:
+        given = {}
+
+    if given and not defaults:
+        msg = "Parameters given, but none declared"
+        raise Exception(msg)
+
+    required = [k for k, v in defaults.items() if v is None]
+    missing = set(required).difference(given.keys())
+    if missing:
+        msg = f"Missing parameters: {', '.join(missing)}"
+        raise Exception(msg)
+
+    extra = set(given.keys()).difference(defaults.keys())
+    if extra:
+        msg = f"Unknown parameters: {', '.join(extra)}"
+        raise Exception(msg)
+
+    ret = defaults.copy()
+
+    for k, v in given.items():
+        if type(v) not in [bool, int, float, str]:
```

Comment:
```suggestion
        if isinstance(v, (int, float, str)):
```

This is kind of subjective, but checking for subtyping is better than type equality.

`IntEnum`/`StrEnum` is I think a reasonable example of subclasses of these types that someone might use here: https://docs.python.org/3/library/enum.html#enum.IntEnum

(For better or worse, `bool` is a subtype of `int`, but feel free to keep it explicitly in the list if you want)

Reply:
yeah, we use `isinstance` elsewhere, I did not know it worked with the tuple of possibilities like that so this is great